### PR TITLE
Add more forge tests

### PR DESCRIPTION
**Pull request by stephankmin**

### CODE REVIEW

File: src/lib/ERC7498NFTRedeemables.sol [hunk 1]
```diff
@@ -182,34 +182,31 @@ contract ERC7498NFTRedeemables is IERC7498, RedeemablesErrors {
...
 
     function _transferConsiderationItem(uint256 id, ConsiderationItem memory c) internal {
+        // WITH_CRITERIA with identifier 0 is wildcard: any id is valid.
+        // Criteria is not yet implemented, for that functionality use the contract offerer.
+        if (
+            id != c.identifierOrCriteria && c.identifierOrCriteria != 0
+                && (c.itemType != ItemType.ERC721_WITH_CRITERIA || c.itemType != ItemType.ERC1155_WITH_CRITERIA)
+        ) {
+            revert InvalidConsiderationTokenIdSupplied(c.token, id, c.identifierOrCriteria);
+        }
```
Comment (ryanio) [inline src/lib/ERC7498NFTRedeemables.sol:192]: nit can you add a newline here

File: test/ERC721ShipyardRedeemable.t.sol [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
-import {RedeemablesErrors} from "../src/lib/RedeemablesErrors.sol";
 import {ERC721RedemptionMintable} from "../src/extensions/ERC721RedemptionMintable.sol";
 import {ERC721ShipyardRedeemableOwnerMintable} from "../src/test/ERC721ShipyardRedeemableOwnerMintable.sol";
 
-contract TestERC721ShipyardRedeemable is RedeemablesErrors, Test {
+contract TestERC721ShipyardRedeemable is BaseRedeemablesTest {
+    using OfferItemLib for OfferItem;
+    using OfferItemLib for OfferItem[];
+    using ConsiderationItemLib for ConsiderationItem;
+    using ConsiderationItemLib for ConsiderationItem[];
```
Comment (ryanio) [inline test/ERC721ShipyardRedeemable.t.sol:22]: are these needed even thouhgh it's in BaseRedeemablesTest?

File: test/ERC721ShipyardRedeemable.t.sol [hunk 3]
```diff
@@ -22,36 +32,105 @@ contract TestERC721ShipyardRedeemable is RedeemablesErrors, Test {
...
+        redeemToken.mint(dillon.addr, invalidTokenId);
+
+        CampaignRequirements[] memory requirements = new CampaignRequirements[](
+            1
+        );
+
+        requirements[0] = CampaignRequirements({
+            offer: defaultCampaignOffer,
+            consideration: defaultCampaignConsideration,
+            traitRedemptions: new TraitRedemption[](0)
```
Comment (ryanio) [inline test/ERC721ShipyardRedeemable.t.sol:93]: can you make a defaultTraitRedemptions as well that's just blank for now

File: test/ERC721ShipyardRedeemable.t.sol (hunk 2 above)
Comment (stephankmin) [reply test/ERC721ShipyardRedeemable.t.sol:22]: yea it won't compile without those lines

looks like libraries aren't inherited starting from [0.7.0](https://docs.soliditylang.org/en/latest/070-breaking-changes.html#functions-and-events)

**Review by ryanio — APPROVED** [review]

**Pull request merged**